jujugui/static/gui/src/app/components/service-overview/test-service-overview.js

```diff
@@ -34,11 +29,131 @@ describe('ServiceOverview', function() {
     YUI().use('service-overview', function() { done(); });
   });
 
-  it('generates a list of actions', function() {
-      var component = renderIntoDocument(
-          <juju.components.ServiceOverview />);
-      assert.isTrue(
-          queryComponentSelector(
-            component, '.overview-action', true).length > 0);
+  it('shows the all units action', function() {
+    var service = {
+      get: function() {
+        return {
+          toArray: function() {
+            return [{}, {}];
+          }
+        };
+      }};
+    var shallowRenderer = testUtils.createRenderer();
+    var icons = juju.components.ServiceOverview.icons;
```

Comment:
Thanks for doing this dance to 'save', 'stub', 'reset' unfortunately if there is a failure in any test that works like this then it will cause cascading failures because the icons object will never get reset. Could you setup an `afterEach` which would be called regardless to get the runner back into a stable state.